src/features/users.ts

```diff
@@ -0,0 +1,34 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { User } from '../types/User';
+import { getUser } from '../api';
+
+type InitialState = {
+  user: User | null;
+  loaded: boolean;
+};
+
+const initialState: InitialState = {
+  user: null,
+  loaded: false,
+};
+
+export const init = createAsyncThunk('user/fetch', (userId: number) => {
+  return getUser(userId);
+});
+
+const userSlice = createSlice({
+  name: 'user',
+  initialState,
+  reducers: {},
+  extraReducers(builder) {
+    builder.addCase(init.pending, state => {
+      return { ...state, loaded: true };
```

Comment:
The `loaded` state should be set to `false` in the `init.pending` case to indicate that data is being fetched. This will ensure that the `Loader` component is displayed during data fetching.